src/server/set_family.cc

```diff
@@ -504,6 +505,8 @@ OpResult<SvArray> InterResultVec(const ResultStringVec& result_vec, unsigned req
 
   for (const auto& k_v : uniques) {
     if (k_v.second == required_shard_cnt) {
+      if (limit != 0 && result.size() >= limit)
```

Comment:
My only concern with this, is that `result.reserve` might allocate memory for a large array while the limit is relatively small (let's say 1-2). Is it worth optimizing or shall we leave it for later on (if it's ever needed we know what to do anyway). 

My two cents are we shouldn't care for now, so @evelez7 feel free to ignore me if @dranikpg agrees as well

Reply:
This was a major concern for me as the redis documentation says to return as soon as we have reached the limit, which is much earlier than this. 

I think there would have to be some refactoring of how the intersection is computed, from my light experimenting. I could work to include this if this is actually a bug according to the redis doc. 

ZINTERCARD also suffers from computing a much larger set than what might be needed.